### DISCUSSION TITLE
rewrite routes to external pages

### DISCUSSION THREAD

**Discussion by biowaffeln** [vercel/vercel#5890] [Help]
Hi! I have a question about rewrites related to #4964. My problem is essentially the same but reverse.
I have a site on vercel (main.vercel.app), and I would like to rewrite different pages to other vercel sites. So what I want is:
```
main.vercel.app/page-a   -->   page-a.vercel.app
main.vercel.app/page-b   -->   page-b.vercel.app
```

When I tried this configuration, it would correctly load the page, but all the assets (page-a.vercel.app/styles.css etc.) would not be loaded:

```json
{
  "rewrites": [
    {
      "source": "/page-a",
      "destination": "https://page-a.vercel.app/"
    }
  ]
}
```

When I tried this configuration, it would redirect all the assets, but not the initial page:
```json
{
  "rewrites": [
    {
      "source": "/page-a(.*)",
      "destination": "https://page-a.vercel.app$1"
    }
  ]
}
```

So what's the correct configuration supposed to be?

**Comment by sanmak** [comment]
@biowaffeln  I'm also looking for a similar solution. Have you found anything yet?

**Comment by fu-sen** [comment]
It may be a bug. This doesn't work:
```
{
  "rewrites": [
    {
      "source": "/test(.*)",
      "destination": "https://example.com"
    }
  ]
}
```
If you change `https://example.com` to` https://example.com/`, it works.

**Comment by sanmak** [comment]
@fu-sen @biowaffeln 
The solution mentioned in the example, [with zones](https://github.com/vercel/next.js/tree/master/examples/with-zones), works for me.

Let's say you have an app [A.vercel.app](A.vercel.app) and an app [B.vercel.app](B.vercel.app). You want to rewrite B.vercel.app/{path} to A.vercel.app/{path}, then you need to update,

- Rewrites config in [B.vercel.app](B.vercel.app). Eg: https://github.com/vercel/next.js/blob/canary/examples/with-zones/home/next.config.js
- And update next config in [A.vercel.app](A.vercel.app). Eg: https://github.com/vercel/next.js/blob/canary/examples/with-zones/blog/next.config.js

It should work with this pattern.
